blockade/cli.py

```diff
@@ -149,7 +149,9 @@ def __with_containers(opts, func):
     configured_containers = set(state.containers.keys())
     container_names = configured_containers if select_all or None else configured_containers.intersection(containers)
 
-    if len(container_names) > 0:
+    if len(container_names) > 0 and hasattr(opts, 'signal'):
+        return func(b, container_names, state, signal=opts.signal)
+    elif len(container_names) > 0:
```

Comment:
nipick you can take or leave: I don't love adding the `signal` check here since this function is used for other commands. Consider having this `_with_containers` function accept `**kwargs` that get passed through to the underlying function. Or use a `functools.partial` in `cmd_kill`.